src/store/categories/reducer.ts

```diff
@@ -0,0 +1,42 @@
+import {getCategories} from '../../api';
+import {Category, LOAD_STATUSES} from '../../types';
+import {createAsyncThunk, createSlice} from "@reduxjs/toolkit";
+
+const CATEGORIES_NAME = 'category';
+
+const fetchCategory = createAsyncThunk(CATEGORIES_NAME, getCategories);
+
+export interface State {
+    categories: Category[];
+    loadStatus: string;
+}
+const initialState: State = {
+    categories: [{id: "", type: "", label: ""}],
```

Comment:
для массивов начально значение - пустой массив, не нужно создавать несуществующие значения
```suggestion
    categories: [],
```